BT CTS fails to build with some libCs

**Describe the bug**
The BT CTS service fails to build with some libCs

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=nrf52_bsim ../samples/bluetooth/peripheral/ 
3. ninja
4. See error

**Expected behavior**
No build failures

**Impact**
Samples fails to build for some targets or with some libCs

**Logs and console output**
```
[97/247] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/bluetooth/services/cts.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/subsys/bluetooth/services/cts.c.obj
[..]
zephyr.dir/subsys/bluetooth/services/cts.c.obj -c subsys/bluetooth/services/cts.c
In file included from /usr/include/sys/types.h:130,
                 from include/zephyr/bluetooth/gatt.h:23,
                 from subsys/bluetooth/services/cts.c:15:
/usr/include/bits/types/timer_t.h:7:19: error: conflicting types for ‘timer_t’; have ‘__timer_t’ {aka ‘void *’}
    7 | typedef __timer_t timer_t;
      |                   ^~~~~~~
In file included from include/zephyr/posix/time.h:61,
                 from subsys/bluetooth/services/cts.c:12:
include/zephyr/posix/posix_types.h:93:23: note: previous declaration of ‘timer_t’ with type ‘timer_t’ {aka ‘long unsigned int’}
   93 | typedef unsigned long timer_t;
      |                       ^~~~~~~
subsys/bluetooth/services/cts.c: In function ‘bt_cts_time_from_unix_ms’:
subsys/bluetooth/services/cts.c:74:13: warning: implicit declaration of function ‘gmtime_r’; did you mean ‘gmtime’? [-Wimplicit-function-declaration]
   74 |         if (gmtime_r(&unix_ts, &date_time) == NULL) {
      |             ^~~~~~~~
      |             gmtime
subsys/bluetooth/services/cts.c:74:44: warning: comparison between pointer and integer
   74 |         if (gmtime_r(&unix_ts, &date_time) == NULL) {
      |                                            ^~
``` 

**Environment (please complete the following information):**

 - OS: Ubuntu Linux 24.04
 - Toolchain (e.g Zephyr SDK, ...): 0.17 or host gcc 14.2
 - Commit SHA or Version used: 8dc3f856229ce083c956aa301c31a23e65bd8cd8

**Additional context**
Related to #81412
Service introduced in 957d59814b29b9ba3e9dcd707c474393baab639b

--- a/include/zephyr/bluetooth/services/cts.h
+++ b/include/zephyr/bluetooth/services/cts.h
@@ -16,7 +16,6 @@
  */
 
 #include <stdint.h>
-#include <zephyr/posix/time.h>
 
 #ifdef __cplusplus
 extern "C" {

--- a/subsys/bluetooth/services/cts.c
+++ b/subsys/bluetooth/services/cts.c
@@ -7,9 +7,17 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+
+#undef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L /* To get gmtime_r()'s prototype */
+
+#ifdef CONFIG_BT_CTS_HELPER_API
+#include <time.h>
+#include <zephyr/sys/timeutil.h>
+#endif
+
 #include <stdbool.h>
 #include <zephyr/sys/byteorder.h>
-#include <zephyr/posix/time.h>
 
 #include <zephyr/bluetooth/conn.h>
 #include <zephyr/bluetooth/gatt.h>
@@ -25,9 +33,6 @@ LOG_MODULE_REGISTER(cts, CONFIG_BT_CTS_LOG_LEVEL);
 static const struct bt_cts_cb *cts_cb;
 
 #ifdef CONFIG_BT_CTS_HELPER_API
-
-#include <time.h>
-#include <zephyr/sys/timeutil.h>
 
 int bt_cts_time_to_unix_ms(const struct bt_cts_time_format *ct_time, int64_t *unix_ms)
 {

